Add basic tools for debugging vms to the readme (#71)
hashicorp/nomad-driver-virt: README.md

virsh list
```
## Debugging a VM

Sometimes things dont go as plan and extra tools are necessary to find the problem.

> Here are some strategies to debug a failing vm:

Reviewer: ```suggestion
Here are some strategies to debug a failing VM:
```

### Connecting to a VM

> By default, cloud images are password protected, by adding a `default_user_password`
> a new password is assigned to the default user of the used distribution: `ubuntu` for ubuntu
> `fedora` for fedora, `root` for alpine... 

Reviewer: ```suggestion
By default, cloud images are password protected, by adding a `default_user_password`
a new password is assigned to the default user of the used distribution (for example, 
`ubuntu` for ubuntu `fedora` for fedora, or `root` for alpine)
```

> By running `virsh console [vm-name]`, a terminal is started inside the vm that will allow an internal inspection of the vm.

Reviewer: ```suggestion
By running `virsh console [vm-name]`, a terminal is started inside the VM that will allow an internal inspection of the VM.
```

```
$ virsh list
 Id   Name                 State
------------------------------------
 1    virt-task-8bc0a63f   running

$ virsh console virt-task-8bc0a63f 
Connected to domain 'virt-task-8bc0a63f'
Escape character is ^] (Ctrl + ])

nomad-virt-task-8bc0a63f login: ubuntu
Password:
```

> If no login promt shows up, it can mean the virtual machine is not booting, adding some extra 
> space to the disk can solve the problem, remember the disk has to fit the cloud image
> plus any other process running in the vm.

Reviewer: ```suggestion
If no login prompt shows up, it can mean the virtual machine is not booting and 
adding  some extra  space to the disk may solve the problem. Remember the disk 
has to fit the root image plus any other process running in the VM.
```

What does "plus any other process" refer to here?

Author: Whatever you might install inside de vm

Reviewer: Isn't that bound by the size of the image's file system? Ex. if `/` is 10GiB, we need to fit 10GiB of space but anything installed during boot is going to be inside that same 10GiB.

The virt driver heavily relies on `cloud-init` to execute the virtual machine's 
configuration. Once you have managed to connect to the terminal, the results of 
cloud init can be found in two different places:
  * `/var/log/cloud-init.log`
  * `/var/log/cloud-init-output.log`

Looking into these files can give a better understanding of any possible execution
errors.

If connecting to the terminal is not an option, it is possible to stop the job and

> mount the vm's disk to inspect it. If the `use_thin_image` option is used, the driver will create 
> the disk image in the directory `plugin_config.data_dir/virt/vm-name.img`:

Reviewer: ```suggestion
mount the VM's disk to inspect it. If the `use_thin_image` option is used, the driver will create 
the disk image in the directory `${plugin_config.data_dir}/virt/vm-name.img`:
```

```

> ubuntu:/var/lib/virt$ ls
> virt-task-8bc0a63f.img
> ```
> 
> ```
> # Enable Network Block Devices on the Host

Reviewer: I'd join this section to the section below:

```suggestion
# Find the virtual machine disk image
$ ls /var/lib/virt
virt-task-8bc0a63f.img

# Enable Network Block Devices on the Host
```

modprobe nbd max_part=8

# Connect the disk as network block device
qemu-nbd --connect=/dev/nbd0 '/var/lib/virt/virt-task-dc8187e3.img'

# Find The Virtual Machine Partitions
fdisk /dev/nbd0 -l

# Mount the partition from the VM
mount /dev/nbd0p1 /mnt/somepoint/
```

**Important** Don't forget to unmount the disk after finishing:

```
umount /mnt/somepoint/
qemu-nbd --disconnect /dev/nbd0
rmmod nbd
```

### Networking

For networking, the plugin leverages on the libvirt default network `default`:
```
$ virsh net-list
 Name      State    Autostart   Persistent
--------------------------------------------
 default   active   yes         yes
 ```
 

> Under the hood, libvirt uses [dnsmask](https://thekelleys.org.uk/dnsmasq/doc.html) to lease 

Reviewer: ```suggestion
Under the hood, libvirt uses [dnsmasq](https://thekelleys.org.uk/dnsmasq/doc.html) to lease 
```
